vault/core.go

```diff
@@ -517,6 +517,8 @@ type Core struct {
 	// clusterListener starts up and manages connections on the cluster ports
 	clusterListener *atomic.Value
 
+	customListenerHeader *ListenersCustomResponseHeadersList
```

Comment:
We'll need to make this field threadsafe, either with a lock or by making it atomic.